Simplified fields

joyal-mathew opened the pull request:
This PR introduces the new simplified field interface.

jacobmerson committed:
remove combiners that are no longer used after gather/scatter removal

jacobmerson committed:
remove non standar library span

We use Kokkos MDSpan for everything now.

jacobmerson committed:
remove unused coordinate array types

jacobmerson committed:
ScalarArrayView -> Rank1View

This refactor was made in preparation for updates that will use Rank2
arrays as well for consistency.

jacobmerson committed:
move field adapters into folder

jacobmerson committed:
remove CoordinateElement type from OmegaH fields

jacobmerson committed:
add field layout and fiel2 that uses layout

joyal-mathew committed:
initial api transition steps

joyal-mathew committed:
rough interpolate_field code

joyal-mathew committed:
refined omega_h evaluation adapter

joyal-mathew committed:
testing field evaluation and copy

joyal-mathew committed:
test interpolate field & "NodalData" -> "DOFHolderData"

joyal-mathew committed:
field communicator implementation and test

joyal-mathew committed:
added profiling statements

joyal-mathew committed:
made communicator layout specific instead of field specific

joyal-mathew committed:
factor out FieldLayoutCommunicator from FieldCommunicator2

joyal-mathew committed:
merge scorec/pcms/develop

joyal-mathew committed:
add meshFields dependency

joyal-mathew committed:
initial field generalization

joyal-mathew committed:
pcms meshFields backend implementation

joyal-mathew committed:
initial generalized field communication

joyal-mathew committed:
field communication update

joyal-mathew committed:
Communication bug fixes

joyal-mathew committed:
initial point cloud adapter implementation

jacobmerson committed:
move header files to headerfile list

jacobmerson committed:
update interpolation tests

The interpolation tests are updated to check the identity interpolation
is working. There was a bunch of extra code that was not needed for these
tests. Note, these tests will not pass in debug mode without adding some
tolerance to the meshFields. See: https://github.com/SCOREC/meshFields/issues/69

jacobmerson committed:
udpate MeshFields

This commit updates the version of MeshFields that PCMS uses to
61e0432152f33568ffe2a96b85495977ac3b2111

Note, the indexing changed in MeshFields 5c0313e33f38e6400f36376729041e5c7cf06f40

jacobmerson committed:
Merge branch 'develop' into merge-simplified-fields

jacobmerson committed:
update test_field_communication to use pcms_testcases data

this commit updates the field_communication test to work with data
in the SCOREC/pcms_testcases repo. The matching commit that adds the needed
data is 53cbd1d0b5732695b9f7d48876dbb70696eb3abb

jacobmerson committed:
cleanup some compiler warnings

jacobmerson committed:
add prefix to macro variables to avoid shadow warnings

jacobmerson commented:
TODO List:

- [ ] https://github.com/SCOREC/meshFields/issues/69 (needed for test_field_communication_4p to pass
- [ ] build with no warnings on new code

jacobmerson committed:
move dummy_field_adapter to adapter folder

jacobmerson committed:
move the coordinates of the DOFHolders to the FieldLayout

As I'm working through the field updates it has become clear that the DOF
coordinates should be part of the field layout. This becomes clear when
you realize that computing the ReversePartitionMap requires the coordinates
and the reverse partition map is a key component of the FieldLayout.
To a large extent, Field's only responsibility should be to know how to do
evaluations. Likely this indicates that the Serialize/Deserialize functionality
should be pushed down to the FieldLayout as well.

jacobmerson committed:
cleanup compiler warnings

jacobmerson commented on src/pcms/adapter/omega_h/omega_h_field_layout.cpp:
```diff
@@ -0,0 +1,252 @@
+
+  auto coords = mesh_.coords();
+
+  size_t offset = 0;
```
https://github.com/SCOREC/meshFields/issues/70

joyal-mathew committed:
initial interface redesign

joyal-mathew committed:
update/simplify new api

- Handle dof holder data using Rank1View
- Make CreateField a method of FieldLayout

jacobmerson commented:
@joyal-mathew Can you resolve the conflicts?

joyal-mathew commented:
Yes, I'll resolve these conflicts today.

joyal-mathew committed:
Merge branch 'main' into simplified-fields

joyal-mathew committed:
formatting update

joyal-mathew committed:
updated .git-blame-ignore-revs

joyal-mathew commented:
It seems like there are some false positives from clang-tidy. It complains about these two cases being the same (`error: switch has 2 consecutive identical branches`):

`pcms/src/pcms/adapter/omega_h/omega_h_field2.cpp:131`

```c
      switch (mesh_.dim()) {
        case 1:
          mesh_field_ = std::make_unique<MeshFieldBackendImpl<1, 1>>(mesh_);
          break;
        case 2:
          mesh_field_ = std::make_unique<MeshFieldBackendImpl<2, 1>>(mesh_);
          break;
      }
```

I think it may be getting confused by the template.

Sichao25 commented on test/test_field_communication.cpp:
```diff
@@ -0,0 +1,209 @@
+  ts::ClassificationPartition cp;
+  std::map<redev::ClassPtn::ModelEnt, int> ent_to_rank;
+
+  for (int e = 0; e <= 0; ++e) {
```
Does this mean that this block will only trigger once. If so,could this be simplified?

Sichao25 reviewed (commented):
Added a few minor questions and suggestions for consideration.

joyal-mathew committed:
cleaned up code

joyal-mathew committed:
Merge branch 'develop' into simplified-fields

joyal-mathew committed:
formatting

joyal-mathew committed:
fix workflow build script

jacobmerson commented:
[0001-update-meshfields-workflow.patch](https://github.com/user-attachments/files/22918628/0001-update-meshfields-workflow.patch)

@joyal-mathew  Can you give this patch a try?

joyal-mathew committed:
applied meshfields workflow patch

jacobmerson commented:
Looks like some minor clang-tidy issues, but the build/tests are back to passing.

joyal-mathew committed:
fixed clang-tidy issues

joyal-mathew committed:
actually fixed all clang-tidy issues
